pointcloud_filter_node:
  ros__parameters:
    
    # Topics
    input_lidar_pointcloud_topic_name: "/point_cloud/raw"
    input_lidar_pointcloud_topic_queue_size: 10
    input_lidar_pointcloud_topic_frame_id: "radar"

    input_camera_pointcloud_topic_name: "/point_cloud/camera/raw"
    input_camera_pointcloud_topic_queue_size: 10
    input_camera_pointcloud_topic_frame_id: "d435i_camera_depth_optical_frame"

    output_pointcloud_topic_name: "/point_cloud/filtered"
    output_pointcloud_topic_queue_size: 10
    output_pointcloud_topic_frame_id: "base_footprint"

    # Parameters
    crop_box:
      x_min: -0.1
      x_max: 0.65
      y_min: -0.25
      y_max: 0.25
      z_min: -0.4
      z_max: 0.25

    ground_plane_segmentation:
      distance_threshold: 0.25 # meters
      max_iterations: 1000
      probability: 0.99

    # Visualization
    visualize_debug: True
    visualization_timer_period: 0.1 # in seconds

    crop_box_visualization_topic_name: "/pointcloud_filter/crop_box_visualization"
    crop_box_visualization_topic_queue_size: 10